src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -794,22 +789,37 @@ static MemberResolutionResult<TMember> makeWrongCallingConvention(MemberResoluti
         }
 #nullable disable
 
-        private bool FailsConstraintChecks(MethodSymbol method, out ArrayBuilder<TypeParameterDiagnosticInfo> constraintFailureDiagnosticsOpt, CompoundUseSiteInfo<AssemblySymbol> template)
+        private bool FailsConstraintChecks<TMember>(TMember member, out ArrayBuilder<TypeParameterDiagnosticInfo> constraintFailureDiagnosticsOpt, CompoundUseSiteInfo<AssemblySymbol> template) where TMember : Symbol
         {
-            if (method.Arity == 0 || method.OriginalDefinition == (object)method)
+            bool isNewExtensionMember = member.GetIsNewExtensionMember();
```

Comment:
>isNewExtensionMember [](http://example.com/codeflow?start=17&length=20)

It looks like the local is used only in one place. Consider inlining or moving the declaration closer to the reference.